platform-validation/scripts/compatibility.sh

```diff
@@ -393,6 +393,39 @@ perform_capability_check_container(){
     wrap_pass "capability_check_container" "Pass"
 }
 
+check_shared_library_dependency(){
+    wrap_debug "Checking shared library dependency for aziot-edged and aziot-identityd"
+
+    # check dependencies 
+    ret=$(need_cmd ldconfig) && $(need_cmd grep)
```

Comment:
I think we can remove our dependency on ldconfig. What i recommend doing is 
if ldconfig is present - use that way
if it is not present -> perform a  ```find . -name "<library-name"``` check in the following directories
1. /usr 
2. /lib
3. /lib32
4. /lib64

At every point, you would need to check if that directory actually exists, if it doesn't move to the next. If it does, perform a find operation and check for both exit code and returned data. If returned data is contains the library name, pass that library check. If its neither, fail and move on.

find command is usually available on most *nix systems